version: '3.9'

services:
  beethoven-db:
    image: postgres:16.0-alpine
    ports:
      - '5434:5432'
    environment:
      POSTGRES_USER: beethoven_user
      POSTGRES_PASSWORD: beethoven_password
      POSTGRES_DB: beethoven_db
    healthcheck:
      test: [ "CMD-SHELL", "pg_isready", "-d", "beethoven_db" ]
      interval: 30s
      timeout: 60s
      retries: 5
      start_period: 10s
    networks:
      - beethoven-localnet
    restart: always

  beethoven:
    image: beethoven:local
    build:
      context: ../
      dockerfile: ./docker/Dockerfile
    command: [ 'run', '--cfg', '/config/beethoven.toml' ]
    depends_on:
      beethoven-db:
        condition: service_healthy
    ports:
      - '8444:8444'
    volumes:
      - ./data/beethoven/beethoven.keystore:/pk/beethoven.keystore
      - ./data/beethoven/beethoven.toml:/config/beethoven.toml
    networks:
      - beethoven-localnet
      - zkevm-localnet
    restart: always

networks:
  beethoven-localnet:
    driver: bridge
  zkevm-localnet:
    name: zkevm-localnet
    external: true
